### ISSUE TITLE
[RELEASE] Release 3.5

### ISSUE THREAD

**Issue by planetf1** [odpi/egeria#6140]
### Work Plan

Create release x.y :

Prior to the release work
- [x] slack post advising of upcoming release & linking to issue
- [x] advance warning in developer/community call
- [x] final agreement to start branch in team call & identification of outstanding issues
- [x] Agree required updates/versions for additional repos including egeria-ui, egeria-reactui, connectors etc

Branching & Correcting versions
- [x] Create branch
- [x] Reassign any issues not being worked on to the next release
- [x] Update version for master (ie x.y-SNAPSHOT > x.y+1-SNAPSHOT)
- [x] Update version for branch (ie x.y-SNAPSHOT -> x.y)

Final updates to the release
- [x] Ensure any remaining fixes are merged into branch (and vice-versa to master)

Generate a release image for testing
- [x] Start release pipeline manually to generate docker image

Updating the React UI (as this is part of the notebook test). Note egeria-ui is out of scope other than inclusion of current release
- [n/a] Create branch
- [n/a] Update version in master
- [n/a] Update version in new branch 
- [n/a] Ensure new docker image is available for testing (there is no distinct release pipeline)

Updating the Helm Charts (egeria-charts repo)
- [x] checker correct docker images are on dockerhub (these are built by the 'merge' build of a release)
- [x] update image versions for helm charts (egeria-charts repo) (using -prerelease for chart version)

Final tests
- [x] Check swagger doc renders (no regressions)
- [x] Verify odpi-egeria-lab chart (pods active/ready)
- [x] Verify egeria-base chart (pods active/ready)
- [x] Check notebooks (config, start, data catalog at a minimum)
- [x] Check polymer UI (only possible to check it runs - no demo scenario for more)
- [x] Check React UI (rex, tex, glossary author)
- [x] CTS - graph
- [x] CTS - inmemory

Final Docs
- [x] Update release notes in egeria-docs

Final build and publish
- [x] Run 'release' pipeline on branch to push candidates to oss.sonatype.org
- [x] 'close' staging repo & Validate artifacts ok (number, structure, validations) on oss.sonatype.org
- [x] Create final github releases for egeria (add link to egeria docs)
- [x] Close repo on oss.sonatype.org (once updated) for egeria, release
- [x] Check 'release' repo on oss.sonatype.org has artifacts
- [x] Update final versions of egeria-charts to release ie x.y
- [x] Publish that release is now shipped via slack #egeria-announce
- [ ] Additional posts to social media
- [x] Communicate to other repo owners ie for connectors so that they can be rebuilt/shipped as needed

Additional components
- ReactUI
  - [n/a] Create final github release
- PostGres Connector
  - [n/a] Create branch
  - [n/a] Update version in master
  - [n/a] Update version in new branch of connector, and of prereq egeria artifacts
  - [n/a] Run release action, check sonatype & release
  - [n/a] Create github release

**Comment by planetf1** [comment]
I will propose we create the release (branch) on Thu 27 Jan, but we will review on the developer
call tomorrow 2022-01-24

**Comment by lpalashevski** [comment]
Yes lets keep the same. There are no functional changes going on at the moment in the polymer version. The ongoing activities are related to migration to React (in the same codebase).

**Comment by planetf1** [comment]
@lpalashevski Ok thanks for confirming, will stick with the same polymer ui version.

**Comment by planetf1** [comment]
* Waiting on https://github.com/odpi/egeria/pull/6150 to be merged - expected this week @mandy-chessell 
* If this PR takes more than 1-1.5 days will branch anyway & cherry-pick the fix to the release
* Will branch sometime this week
* Sticking with Dec release of ecosystem UI (ie in charts) now due to some changes around enabling access services @davidradl 
* Will branch ecosystem UI when code changes there complete, and then update charts again (ie to 3.5.1)

**Comment by planetf1** [comment]
inmemory CTS complete:
```
	name	description	profilePriority	conformanceStatus
0	Metadata sharing	The technology under test is able to share met...	MANDATORY_PROFILE	CONFORMANT_FULL_SUPPORT
1	Reference copies	The technology under test is able to save, loc...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
2	Metadata maintenance	The technology under test supports requests to...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
3	Dynamic types	The technology under test supports changes to ...	OPTIONAL_PROFILE	UNKNOWN_STATUS
4	Graph queries	The technology under test supports graph-like ...	OPTIONAL_PROFILE	CONFORMANT_PARTIAL_SUPPORT
5	Historical search	The technology under test supports search for ...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
6	Entity proxies	The technology under test is able to store stu...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
7	Soft-delete and restore	The technology under test allows an instance t...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
8	Undo an update	The technology under test is able to restore a...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
9	Reidentify instance	The technology under test supports the command...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
10	Retype instance	The technology under test supports the command...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
11	Rehome instance	The technology under test supports the command...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
12	Entity search	The technology under test supports the ability...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
13	Relationship search	The technology under test supports the ability...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
14	Entity advanced search	The technology under test supports the use of ...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
```
```
        "testCaseCount": 5612,
        "testFailedCount": 0,
        "testPassCount": 5612,
```

**Comment by planetf1** [comment]
Graph CTS complete:
```
	name	description	profilePriority	conformanceStatus
0	Metadata sharing	The technology under test is able to share met...	MANDATORY_PROFILE	CONFORMANT_FULL_SUPPORT
1	Reference copies	The technology under test is able to save, loc...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
2	Metadata maintenance	The technology under test supports requests to...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
3	Dynamic types	The technology under test supports changes to ...	OPTIONAL_PROFILE	UNKNOWN_STATUS
4	Graph queries	The technology under test supports graph-like ...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
5	Historical search	The technology under test supports search for ...	OPTIONAL_PROFILE	CONFORMANT_NO_SUPPORT
6	Entity proxies	The technology under test is able to store stu...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
7	Soft-delete and restore	The technology under test allows an instance t...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
8	Undo an update	The technology under test is able to restore a...	OPTIONAL_PROFILE	CONFORMANT_NO_SUPPORT
9	Reidentify instance	The technology under test supports the command...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
10	Retype instance	The technology under test supports the command...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
11	Rehome instance	The technology under test supports the command...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
12	Entity search	The technology under test supports the ability...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
13	Relationship search	The technology under test supports the ability...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
14	Entity advanced search	The technology under test supports the use of ...	OPTIONAL_PROFILE	CONFORMANT_FULL_SUPPORT
```
```
        "testCaseCount": 5612,
        "testFailedCount": 0,
        "testPassCount": 5612,
```

**Comment by planetf1** [comment]
@wbittles @mandy-chessell I am proposing not to release the Postgres connector as there have been no updates this release cycle. Are you both ok with that?

**Comment by mandy-chessell** [comment]
@planetf1 yes

**Comment by planetf1** [comment]
@cmgrote @wbittles FYI in case you wish to make any new release updates to additional connectors. Let me know if you wanted & need any help
@davidradl @lpalashevski Confirming the release is out with ecoui 3.4.0 & genui 3.2.0 - if updates needed before next egeria release, please open issue/PR in the egeria-charts repo
